File: app/src/main/java/it/amonshore/comikkua/ui/OnNavigationFragmentListener.java
package it.amonshore.comikkua.ui;

import android.net.Uri;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.appcompat.view.ActionMode;
import it.amonshore.comikkua.ICallback;

public interface OnNavigationFragmentListener {

    void onFragmentInteraction(Uri uri);

    void onFragmentRequestActionMode(@Nullable ActionMode.Callback callback, String name, CharSequence title);
}

File: app/src/main/java/it/amonshore/comikkua/ui/OnNavigationFragmentListener.kt
package it.amonshore.comikkua.ui

import androidx.appcompat.view.ActionMode
import java.time.Duration

interface OnNavigationFragmentListener {
    fun onFragmentRequestActionMode(
        name: String,
        title: CharSequence? = null,
        callback: ActionMode.Callback? = null
    )

    fun handleUndo(message: String, tag: String, timeout: Duration = Duration.ZERO)

    fun resetUndo()
}